pkg/gate/gate.go

```diff
@@ -26,17 +27,22 @@ func New(delegate http.Handler, pause, resume func()) http.HandlerFunc {
 			select {
 			case <-doneCh:
 				inFlight--
+				fmt.Println("inFlight after request",inFlight)
```

Comment:
I think if we want to log here we should use a proper logger, but either way let's keep this PR focused on one thing and do any changes to this file separately?

Reply:
dropped